streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -74,31 +77,48 @@ public AssignmentInfo(final int version,
                           final List<TaskId> activeTasks,
                           final Map<TaskId, Set<TopicPartition>> standbyTasks,
                           final Map<HostInfo, Set<TopicPartition>> hostState) {
-        this(version, LATEST_SUPPORTED_VERSION, activeTasks, standbyTasks, hostState);
+        this(version, LATEST_SUPPORTED_VERSION, activeTasks, standbyTasks, hostState, 0);
 
         if (version < 1 || version > LATEST_SUPPORTED_VERSION) {
             throw new IllegalArgumentException("version must be between 1 and " + LATEST_SUPPORTED_VERSION
                 + "; was: " + version);
         }
     }
 
+    public AssignmentInfo(final int version,
+        final List<TaskId> activeTasks,
+        final Map<TaskId, Set<TopicPartition>> standbyTasks,
+        final Map<HostInfo, Set<TopicPartition>> hostState,
+        final int errCode) {
+        this(version, LATEST_SUPPORTED_VERSION, activeTasks, standbyTasks, hostState, errCode);
+        if (version < 1 || version > LATEST_SUPPORTED_VERSION) {
```

Comment:
We can remove the code block line 82-85 above since it will be called here.

Reply:
Unfortunately no.
The `this` call goes to line 100 where this is no such check.